hooks/pre/30-el8_upgrade_postgresql.rb

```diff
@@ -0,0 +1,26 @@
+def postgresql_upgrade(new_version)
+  stop_services
+
+  execute!("dnf module switch-to postgresql:#{new_version} -y")
+
+  server_packages = ['postgresql', 'postgresql-server', 'postgresql-upgrade']
+  ['postgresql-contrib', 'postgresql-docs'].each do |extra_package|
+    if execute("rpm -q #{extra_package}", false, false)
+      server_packages << extra_package
+    end
+  end
+
+  ensure_packages(server_packages, 'latest')
+
+  execute!("sed -i '/^data_directory/d' /var/lib/pgsql/data/postgresql.conf")
+
+  execute!("runuser -l postgres -c 'postgresql-setup --upgrade'")
+
+  start_services(['postgresql'])
+
+  execute!("runuser -l postgres -c 'vacuumdb --all --analyze-in-stages'")
+end
+
+if local_postgresql? && el8? && foreman_server? && needs_postgresql_upgrade?(12)
+  postgresql_upgrade(13)
```

Comment:
If `needs_postgresql_upgrade` is updated to only look for the new version this could be:
```suggestion
target_postgresql_version = 13
if local_postgresql? && el8? && foreman_server? && needs_postgresql_upgrade?(target_postgresql_version)
  postgresql_upgrade(target_postgresql_version)
```

And then the target version can be stored